### PR TITLE
Stopgap: MongoDB Capped Collections

### PR DESCRIPTION
**Pull request by Druue**
closes https://github.com/prisma/schema-team/issues/405

contributes to: https://github.com/prisma/prisma/issues/17464

### CODE REVIEW

File: schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs [hunk 1]
```diff
@@ -153,6 +153,11 @@ impl<'a> Statistics<'a> {
                         let comment = "This collection uses a JSON Schema defined in the database, which requires additional setup for migrations. Visit https://pris.ly/d/mongodb-json-schema for more info.";
                         model.documentation(comment)
                     }
+
+                    if walker.is_capped() {
+                        let comment = "This collection is marked as capped, which requires [a good message here]. Visit https://pris.ly/d/mongodb-capped-collections for more info";
```
Comment (Druue) [inline schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs:158]: If anyone has thoughts here on how to word this.
My initial thought felt maybe a little off?
```suggestion
                        let comment = "This collection is marked as capped, which requires further support in Prisma Client. Visit https://pris.ly/d/mongodb-capped-collections for more info";
```

File: schema-engine/connectors/schema-connector/src/warnings.rs [hunk 2]
```diff
@@ -416,6 +418,12 @@ impl fmt::Display for Warnings {
             f
         )?;
 
+        render_warnings(
+            "The following models are marked as capped collections, which are not yet fully supported. Read more: https://pris.ly/d/mongodb-capped-collections",
```
Comment (Druue) [inline schema-engine/connectors/schema-connector/src/warnings.rs:422]: CLI Warning

File: schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs (hunk 1 above)
Comment (janpio) [reply schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs:158]: Maybe like this?
```suggestion
                        let comment = "This model is a capped collection, which requires manual setup when creating the collection in MongoDB: https://pris.ly/d/mongodb-capped-collections";
```

File: schema-engine/connectors/schema-connector/src/warnings.rs (hunk 2 above)
Comment (janpio) [reply schema-engine/connectors/schema-connector/src/warnings.rs:422]: (Equivalent to what we choose above I guess)

File: schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs (hunk 1 above)
Comment (Druue) [reply schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs]: Hm, it feels like it shouldn't necessarily just be specific to creation of the collection in Mongo though. Like it needs both the manual mongo mql and the client work-flow with `runCommandRaw`

File: schema-engine/connectors/schema-connector/src/warnings.rs (hunk 2 above)
Comment (Druue) [reply schema-engine/connectors/schema-connector/src/warnings.rs:422]: Does this one need to change? I thought this one was fine as is

File: schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs (hunk 1 above)
Comment (janpio) [reply schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs]: That is true. Probably something super generic "which requires additional attention to be usable with Prisma" maybe.

File: schema-engine/connectors/schema-connector/src/warnings.rs (hunk 2 above)
Comment (janpio) [reply schema-engine/connectors/schema-connector/src/warnings.rs:422]: I find "marked as" a bit weird, as we use "is" in the other one.
But yeah, does not matter to much.

Comment (Druue) [reply schema-engine/connectors/schema-connector/src/warnings.rs:422]:
```suggestion
        render_warnings(
            "The following models are capped collections, which are not yet fully supported. Read more here: https://pris.ly/d/mongodb-capped-collections",
```

File: schema-engine/connectors/mongodb-schema-connector/tests/introspection/basic/mod.rs [hunk 3]
```diff
@@ -118,3 +118,37 @@ fn collection_with_json_schema() {
...
+    let expected_warning = expect![[r#"
+        *** WARNING ***
+
+        The following models are marked as capped collections, which are not yet fully supported. Read more: https://pris.ly/d/mongodb-capped-collections
+          - "A"
+    "#]];
+    res.expect_warnings(&expected_warning);
+
+    let expected_doc = expect![[r#"
+        /// This collection is marked as capped, which requires [a good message here]. Visit https://pris.ly/d/mongodb-capped-collections for more info
```
Comment (janpio) [inline schema-engine/connectors/mongodb-schema-connector/tests/introspection/basic/mod.rs:148]: TODO

File: schema-engine/connectors/schema-connector/src/warnings.rs (hunk 2 above)
Comment (Druue) [reply schema-engine/connectors/schema-connector/src/warnings.rs:422]: I just realised if we updated this to read as `Read more here: ...` that would be inconsistent with the other warnings but I honestly prefer including the here so am inclined to update the other messages to have it too, thoughts?

Comment (Druue) [reply schema-engine/connectors/schema-connector/src/warnings.rs]: It is currently as follows:
```
The following models are capped collections, which are not yet fully supported. Read more: https://pris.ly/d/mongodb-capped-collections
```

File: schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs (hunk 1 above)
Comment (Druue) [reply schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs]: Updated this as follows
```suggestion
let comment = "This model is a capped collection, which requires further support to be used with Prisma: https://pris.ly/d/mongodb-capped-collections";```

File: schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs [hunk 4]
```diff
@@ -153,6 +153,11 @@ impl<'a> Statistics<'a> {
                         let comment = "This collection uses a JSON Schema defined in the database, which requires additional setup for migrations. Visit https://pris.ly/d/mongodb-json-schema for more info.";
                         model.documentation(comment)
                     }
+
+                    if walker.is_capped() {
+                        let comment = "This model is a capped collection, which requires further support to be used with Prisma: https://pris.ly/d/mongodb-capped-collections";
```
Comment (janpio) [inline schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs:158]:
```suggestion
                        let comment = "This model is a capped collection, which is not yet fully supported. Read more: https://pris.ly/d/mongodb-capped-collections";
```

File: schema-engine/connectors/mongodb-schema-connector/tests/introspection/basic/mod.rs [hunk 5]
```diff
@@ -118,3 +118,37 @@ fn collection_with_json_schema() {
...
+    let expected_warning = expect![[r#"
+        *** WARNING ***
+
+        The following models are capped collections, which are not yet fully supported. Read more: https://pris.ly/d/mongodb-capped-collections
+          - "A"
+    "#]];
+    res.expect_warnings(&expected_warning);
+
+    let expected_doc = expect![[r#"
+        /// This model is a capped collection, which requires further support to be used with Prisma: https://pris.ly/d/mongodb-capped-collections
```
Comment (janpio) [inline schema-engine/connectors/mongodb-schema-connector/tests/introspection/basic/mod.rs:148]:
```suggestion
        /// This model is a capped collection, which is not yet fully supported. Read more: https://pris.ly/d/mongodb-capped-collections
```

File: schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs (hunk 4 above)
Comment (Jolg42) [reply schema-engine/connectors/mongodb-schema-connector/src/sampler/statistics.rs:158]: This is easier to understand to me

**Review by Jolg42 — APPROVED** [review]
✨

**Pull request merged**